x-pack/legacy/plugins/ml/public/data_frame/pages/transform_management/components/transform_list/expanded_row_preview_pane.tsx

```diff
@@ -99,13 +99,44 @@ export const ExpandedRowPreviewPane: FC<Props> = ({ transformConfig }) => {
         if (resp.preview.length > 0) {
           const columnKeys = getFlattenedFields(resp.preview[0]);
           columnKeys.sort(sortColumns(groupByArr));
-          const tableColumns = columnKeys.map(k => {
-            return {
+
+          const tableColumns: FieldDataColumnType[] = columnKeys.map(k => {
+            const column: FieldDataColumnType = {
               field: k,
               name: k,
               sortable: true,
               truncateText: true,
             };
+
+            if (typeof resp.mappings.properties[k] !== 'undefined') {
```

Comment:
This looks like to be largely a copy of the code in `pivot_preview.tsx`. Could the the column handling be moved into a common file for use by the preview table and the table in the expanded row?

Reply:
For now it feels alright to keep them separate. Sounds like a good suggestion for 7.5 if that sounds okay 👍